rest/rest-cli/src/test/java/functionaltests/TagCommandsFunctTest.java

```diff
@@ -366,4 +366,30 @@ public void testListJobResultUnknownJob() throws Exception {
         System.out.println(out);
         assertTrue(out.contains("error"));
     }
+
+    @Test
+    public void testCreateNodeSourceAndListNodeSources() throws Exception {
+        String nodeSourceName = "nodesourcecreationthroughcli";
+        typeLine("createns( '" + nodeSourceName +
+                 "', ['org.ow2.proactive.resourcemanager.nodesource.infrastructure.LocalInfrastructure', '/home/justine/source/repository/fork/scheduling/config/authentication/admin.cred', 4, 60000, ''], ['org.ow2.proactive.resourcemanager.nodesource.policy.StaticPolicy', 'ALL', 'ALL'])");
+
+        runCli();
+
+        String out = this.capturedOutput.toString();
+        System.setOut(stdOut);
+        System.out.println("------------- testCreateNodeSourceAndListNodeSources:");
+        System.out.println(out);
+
+        Thread.sleep(5000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)